src/test/java/com/ably/kafka/connect/offset/OffsetRegistryServiceTest.java

```diff
@@ -0,0 +1,114 @@
+                new SinkRecord("topic1", 1, null, null, null, null, 1),
+                new SinkRecord("topic1", 1, null, null, null, null, 0),
+
+                // Partition 0
+                new SinkRecord("topic1", 0, null, null, null, null, 0),
+
+                // Topic 2 - Partition 0
+                new SinkRecord("topic2", 0, null, null, null, null, 44444),
+                new SinkRecord("topic2", 0, null, null, null, null, 1),
+
+                // Topic 2 - Partition 4
+                new SinkRecord("topic2", 4, null, null, null, null, 22222),
+                new SinkRecord("topic2", 4, null, null, null, null, 1),
+                new SinkRecord("topic2", 4, null, null, null, null, 22223)
+        );
+
+        offsetRegistryService.updateTopicPartitionToOffsetMap(records);
+
+        Map<TopicPartition, Long> expectedResult = new HashMap<>();
+        expectedResult.put(new TopicPartition("topic1", 0), 3L);
+        expectedResult.put(new TopicPartition("topic1", 1), 2L);
+        expectedResult.put(new TopicPartition("topic2", 0), 44444L);
+        expectedResult.put(new TopicPartition("topic2", 4), 22223L);
+
+        offsetRegistryService.getTopicPartitionToOffsetMap().equals(expectedResult);
+
+    }
+
+    @Test
+    public void testUpdateOffsets() {
+        OffsetRegistryService offsetRegistryService = new OffsetRegistryService();
+
+        // There will be offsets of topics
+        // that the sink task will not be subscribed to.
+        Map<TopicPartition, OffsetAndMetadata> preCommitOffsets = new HashMap<>();
+        preCommitOffsets.put(new TopicPartition("topic2", 2323), new OffsetAndMetadata(2200L));
+        preCommitOffsets.put(new TopicPartition("topic2", 2323), new OffsetAndMetadata(1L));
+        preCommitOffsets.put(new TopicPartition("topic2", 2323), new OffsetAndMetadata(2L));
+
+        // Topics that the sink task is subscribed to.
+        preCommitOffsets.put(new TopicPartition("topic1", 0), new OffsetAndMetadata(1L));
+        preCommitOffsets.put(new TopicPartition("topic1", 1), new OffsetAndMetadata(44L));
+
+        List<SinkRecord> records = List.of(
+                // Partition 0
+                new SinkRecord("topic1", 0, null, null, null, null, 3),
+
+                // Partition 1
+                new SinkRecord("topic1", 1, null, null, null, null, 2),
+                new SinkRecord("topic1", 1, null, null, null, null, 1),
+                new SinkRecord("topic1", 1, null, null, null, null, 0)
+                );
+
+        offsetRegistryService.updateTopicPartitionToOffsetMap(records);
+
+        Map<TopicPartition, OffsetAndMetadata> result = offsetRegistryService.updateOffsets(preCommitOffsets);
+
+        Map<TopicPartition, Long> expectedResult = new HashMap<>();
+        expectedResult.put(new TopicPartition("topic1", 0), 3L);
+        expectedResult.put(new TopicPartition("topic1", 1), 2L);
+
+        result.equals(expectedResult);
```

Comment:
This is not a test assertion at the end of this function, you're just calling `Object.equals()` and discarding the result. If you use a test assertion, I think you'll find it fails, because:

* `result` and `expectedResult` don't even have the same type
* One that's fixed, you'll see the bug mentioned above in `updateOffsets`